sway-lib-std/src/hash.sw

```diff
@@ -124,3 +124,31 @@ pub fn hash_pair(value_a: b256, value_b: b256, method: HashMethod) -> b256 {
         }
     }
 }
+
+// Registers are 64 bits (8 bytes) wide. Words are the same width as registers.
+pub fn sha256<T>(param: T) -> b256 {
+    let result_buffer: b256 = 0x0000000000000000000000000000000000000000000000000000000000000000;
+    if !is_reference_type::<T>() {
+        asm(r1: param, r2, r3: 8, r4: result_buffer) {
+            // Move to register r2
+            move r2 sp;
```

Comment:
minor nit, but maybe format with comments after the op to align with pre-existing functions in the module. ie: `move r2 sp; // Move stack pointer to register r2`